ovos_plugin_manager/templates/tts.py

```diff
@@ -453,19 +304,94 @@ def validate_ssml(self, utterance):
         # return text with supported ssml tags only
         return utterance.replace("  ", " ")
 
-    def _preprocess_sentence(self, sentence):
-        """Default preprocessing is no preprocessing.
+    # init helpers
+    def _init_g2p(self):
+        cfg = Configuration()
+        g2pm = self.config.get("g2p_module")
+        if g2pm:
+            if g2pm in find_g2p_plugins():
+                cfg.setdefault("g2p", {})
+                globl = cfg["g2p"].get("module") or g2pm
+                if globl != g2pm:
+                    LOG.info(f"TTS requested {g2pm} explicitly, ignoring global module {globl} ")
+                cfg["g2p"]["module"] = g2pm
+            else:
+                LOG.warning(f"TTS selected {g2pm}, but it is not available!")
 
-        This method can be overridden to create chunks suitable to the
-        TTS engine in question.
+        try:
+            self.g2p = OVOSG2PFactory.create(cfg)
+        except:
+            LOG.debug("G2P plugin not loaded, there will be no mouth movements")
+            self.g2p = None
+
+    def init(self, bus=None, playback=None):
+        """ Performs intial setup of TTS object.
 
         Arguments:
-            sentence (str): sentence to preprocess
+            bus:    OpenVoiceOS messagebus connection
+        """
+        self.bus = bus or BUS()
+        if playback is None:
+            LOG.warning("PlaybackThread should be inited by ovos-audio, initing via plugin has been deprecated, "
+                        "please pass playback=PlaybackThread() to TTS.init")
```

Comment:
Isn't there a decorator for deprecation warnings?

Reply:
there is for functions, but in this case it doesnt work because we only want to log on a class instantiation